huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go

```diff
@@ -0,0 +1,291 @@
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceDdmFlavorsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getDdmFlavors: Query the List of DDM flavors
+	var (
+		getDdmFlavorsHttpUrl = "v2/{project_id}/flavors"
+		getDdmFlavorsProduct = "ddm"
+	)
+	getDdmFlavorsClient, err := cfg.NewServiceClient(getDdmFlavorsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DdmFlavors Client: %s", err)
+	}
+
+	getDdmFlavorsPath := getDdmFlavorsClient.Endpoint + getDdmFlavorsHttpUrl
+	getDdmFlavorsPath = strings.ReplaceAll(getDdmFlavorsPath, "{project_id}", getDdmFlavorsClient.ProjectID)
+
+	getDdmFlavorsQueryParams := buildGetDdmFlavorsQueryParams(d, 0)
+	getDdmFlavorsPath += getDdmFlavorsQueryParams
+	getInstanceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+
+	cpuArch := d.Get("cpu_arch").(string)
+	code := d.Get("code").(string)
+	var vcpus string
+	if v, ok := d.GetOk("vcpus"); ok {
+		vcpus = strconv.Itoa(v.(int))
+	}
+	var memory string
+	if v, ok := d.GetOk("memory"); ok {
+		memory = strconv.Itoa(v.(int))
+	}
+
+	x86Flavors := make([]interface{}, 0)
+	armFlavors := make([]interface{}, 0)
+	for {
+		getDdmFlavorsResp, err := getDdmFlavorsClient.Request("GET", getDdmFlavorsPath, &getInstanceOpt)
+		if err != nil {
+			return common.CheckDeletedDiag(d, err, "error retrieving DdmFlavors")
+		}
+		getDdmFlavorsRespBody, err := utils.FlattenResponse(getDdmFlavorsResp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		res := flattenGetFlavorsResponseBodyFlavorGroup(&queryParma{
+			resp:    getDdmFlavorsRespBody,
+			cpuArch: cpuArch,
+			code:    code,
+			vcpus:   vcpus,
+			memory:  memory,
+		})
+		x86Flavors = append(x86Flavors, res.x86Flavors...)
+		armFlavors = append(armFlavors, res.armFlavors...)
+		if res.offset+res.limit >= res.x86Total && res.offset+res.limit >= res.armTotal {
+			break
+		}
+		getDdmFlavorsPath = updatePathOffset(getDdmFlavorsPath, res.offset+res.limit)
+	}
+
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	var flavors []interface{}
+	flavors = append(flavors, x86Flavors...)
+	flavors = append(flavors, armFlavors...)
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("flavors", flavors),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenGetFlavorsResponseBodyFlavorGroup(parma *queryParma) *queryRes {
+	if parma.resp == nil {
+		return &queryRes{}
+	}
+	curJson := utils.PathSearch("computeFlavorGroups", parma.resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+
+	var x86Flavors []interface{}
+	var armFlavors []interface{}
+	var offset, limit, x86Total, armTotal float64
+
+	for _, v := range curArray {
+		flavorCPUArch := utils.PathSearch("groupType", v, nil)
```

Comment:
```suggestion
		flavorCPUArch := utils.PathSearch("groupType", v, “”)
```
should the  default value be ""?